### PR TITLE
docs(NODE-1456): update configuration documentation

### PR DESCRIPTION
**Pull request by andrewbattat**
As I created the [Configuration revamp doc](https://docs.google.com/document/d/1_apyCGBVZLYvhxG8YrjP06oPMH4MqS4KMT7b5SBbBOw/edit#heading=h.arjp6xabppb4), I saw some out of date documentation.

### CODE REVIEW

File: ic-os/docs/ConfigStore-SetupOSHostOS.adoc
Comment (andrewbattat) [inline ic-os/docs/ConfigStore-SetupOSHostOS.adoc:1]: This file has a lot of unnecessary information. I collapsed the important parts into Configuration.adoc

File: ic-os/components/hostos-scripts/build-bootstrap-config-image.sh [hunk 1]
```diff
@@ -75,8 +75,8 @@ options may be specified:
     keys for specific user accounts on the machine. The name of the
     key designates the name of the account (so, if there is a file
     "PATH/admin" then it is transferred to "~admin/.ssh/authorized_keys" on
-    the target). The presently recognized accounts are: backup, readonly,
-    admin and root (the latter one for testing purposes only!)
+    the target). The presently recognized accounts are: backup, readonly and
+    admin
```
Comment (andrewbattat) [inline ic-os/components/hostos-scripts/build-bootstrap-config-image.sh:79]: We no longer have root ssh access 👍  And apparently I accidentally removed an oxford comma 🤦‍♂️

File: ic-os/docs/Configuration.adoc [hunk 2]
```diff
@@ -29,18 +36,14 @@ GuestOS then reads from `ic-bootstrap.tar` (link:../components/init/bootstrap-ic
 * *System state:* Node initialization material (initial registry & keys) is copied directly into target directories for replica use.
 * *Config partition:* All other contents of the injected config are copied into the `/boot/config` directory. These configurations need to be available on subsequent boots (not just the first boot of the system), and need to be preserved across upgrades.
 ** The contents of `/boot/config` are then used in various ways to parameterize system services as they start. For example, ssh keys are installed to target accounts, network settings are applied by network set-up scripts, etc.
-** For the IC, the `generate-replica-config.sh` script uses the `ic.json5.template` file to produce the configuration required by the replica (also cf. service file definition). Template placeholders are populated with information from `/boot/config` and elsewhere.
+** `generate-replica-config.sh` uses `ic.json5.template` to produce the replica configuration (also cf. service file definition). Template placeholders are populated with information from `/boot/config` and elsewhere.
```
Comment (Bownairo) [inline ic-os/docs/Configuration.adoc:39]: > and elsewhere

😬

**Review by Bownairo — APPROVED** [review]

File: ic-os/docs/Configuration.adoc (hunk 2 above)
Comment (andrewbattat) [reply ic-os/docs/Configuration.adoc]: I just double checked and from what I can see template placeholders are populated with information from `/boot/config` so I removed the 'and elsewhere'

**Pull request merged**